lib/flat_search/flats.ex

```diff
@@ -3,6 +3,28 @@ defmodule FlatSearch.Flats do
   Flats context
   """
   alias FlatSearch.{Repo, Flats.Flat}
+  import Ecto.Query
+
+  @topic inspect(__MODULE__)
+
+  def subscribe do
+    Phoenix.PubSub.subscribe(FlatSearch.PubSub, @topic)
+  end
+
+  def broadcast({:ok, record}, event) do
+    Phoenix.PubSub.broadcast(FlatSearch.PubSub, @topic, {event, record})
+    {:ok, record}
+  end
+
+  def broadcast({:error, _} = error, _event), do: error
+
+  defp notify_subscribers({:ok, result}, event) do
+    Phoenix.PubSub.broadcast(FlatSearch.PubSub, @topic, {__MODULE__, event, result})
+
+    {:ok, result}
+  end
+
+  defp notify_subscribers({:error, reason}, _), do: {:error, reason}
```

Comment:
Maybe this should be moved to another module since this goes beyond the context responsibilities of business logic and working with data